packages/client/src/__tests__/dataTriggers.test.ts

```diff
@@ -674,4 +675,177 @@ describe("Cloud functions -> Data triggers ->", () => {
       }
     );
   });
+  describe("createCustomerStats", () => {
+    testWithEmulator(
+      "should update the customer doc with bookingStats with respect to customer's bookedSlots",
+      async () => {
+        const { organization } = await setUpOrganization();
+        const dateThisMonth = DateTime.now().toISODate();
+        const dateNextMonth = DateTime.now().plus({ month: 1 }).toISODate();
+        const slotThisMonthIce = {
+          ...baseSlot,
+          date: dateThisMonth,
+          type: SlotType.Ice,
+        };
+        const slotNextMonthIce = {
+          ...baseSlot,
+          date: dateNextMonth,
+          type: SlotType.Ice,
+        };
+        const slotThisMonthOffIce = {
+          ...baseSlot,
+          date: dateThisMonth,
+          type: SlotType.OffIce,
+        };
+        const slotNextMonthOffIce = {
+          ...baseSlot,
+          date: dateNextMonth,
+          type: SlotType.OffIce,
+        };
+        // Create customer and slot
+        await Promise.all([
+          adminDb
+            .doc(getSlotDocPath(organization, `${dateThisMonth}-9`))
+            .set(slotThisMonthIce),
+          adminDb
+            .doc(getSlotDocPath(organization, `${dateNextMonth}-9`))
+            .set(slotNextMonthIce),
+          adminDb
+            .doc(getSlotDocPath(organization, `${dateThisMonth}-10`))
+            .set(slotThisMonthOffIce),
+          adminDb
+            .doc(getSlotDocPath(organization, `${dateNextMonth}-10`))
+            .set(slotNextMonthOffIce),
+          adminDb.doc(getCustomerDocPath(organization, saul.id)).set(saul),
+        ]);
+
+        await Promise.all([
+          await waitFor(async () => {
```

Comment:
This `await waitFor` is unnecessary as `waitFor`returns a promise, one which we can `await`, however, `Promise.all` already awaits all promises in the array.